### PR TITLE
chore(runtime): remove builder in core/runtime

### PR DESCRIPTION
**Pull request by lahabana**
Use `opt` which is more Go idiomatic, this also
avoids having to add a new field in multiple places whenever we have a new one

Signed-off-by: Charly Molter <charly.molter@konghq.com>

### Checklist prior to review

- [x] Link to docs PR or issue --
- [x] Link to UI issue or PR --
- [x] Is the [issue worked on linked][1]? --
- [x] The PR does not hardcode values that might break projects that depend on kuma (e.g. "kumahq" as a image registry) --
- [x] The PR will work for both Linux and Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [x] Unit Tests --
- [x] E2E Tests --
- [x] Manual Universal Tests --
- [x] Manual Kubernetes Tests --
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? --
- [x] Do you need to explicitly set a [`> Changelog:` entry here](../blob/master/CONTRIBUTING.md#submitting-a-patch)?

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

**Review by michaelbeaumont — APPROVED** [review]

File: pkg/core/runtime/runtime.go [hunk 1]
```diff
@@ -62,7 +62,7 @@ type RuntimeContext interface {
 	Metrics() metrics.Metrics
 	EventReaderFactory() events.ListenerFactory
 	APIInstaller() api_server.APIInstaller
-	XDSHooks() *xds_hooks.Hooks
+	XDSHooks() []xds_hooks.ResourceSetHook
```
Comment (jakubdyszkiewicz) [inline pkg/core/runtime/runtime.go:65]: I'd prefer to keep it as is. I actually use this object in the external plugin PoC (another type of hooks).

**Review by jakubdyszkiewicz — COMMENTED** [review]
I'm ok with merging this, but if we are dropping builder and going this way, woudn't it be better to change `runtimeContext` to `MutableRuntime` with public fields, pass the pointer and drop all the `WithX`?

**Comment by michaelbeaumont** [comment]
@lahabana Do you still want to get this in?

**Comment by lahabana** [comment]
Doesn't feel like there's unanimity on this so going to close this

**Pull request closed without merging**